src/classes/groundplaces.ts

```diff
@@ -539,48 +660,132 @@ export class GroundPlacesController {
+      );
+    }
+
+    try {
+      groundPlacesDiff.map((groundPlaceDiff: GroundPlaceDiff) => {
+        const [groundPlaceGpuid, { type, into: intoGroundPlaceGpuid, from: fromGroundPlaceGpuid, params }]: [
+          Gpuid,
+          GroundPlaceDiffOptions,
+        ] = Object.entries(groundPlaceDiff)[0];
+
+        switch (type) {
+          case ActionType.CREATE_STOP_GROUP:
+            return this.createStopGroup(params.segmentProviderStopId, groundPlaceGpuid, {
+              countryCode: params.countryCode,
+              latitude: params.latitude,
+              longitude: params.longitude,
+              name: params.name,
+            });
+
+          case ActionType.CREATE_STOP_CLUSTER:
+            return this.createStopCluster(groundPlaceGpuid, {
+              countryCode: params.countryCode,
+              latitude: params.latitude,
+              longitude: params.longitude,
+              name: params.name,
+            });
+
+          case ActionType.UPDATE_STOP_GROUP:
+            return this.updateStopGroup(groundPlaceGpuid, {
+              latitude: params.latitude,
+              longitude: params.longitude,
+              name: params.name,
+            });
+
+          case ActionType.UPDATE_STOP_CLUSTER:
+            return this.updateStopCluster(groundPlaceGpuid, {
+              latitude: params.latitude,
+              longitude: params.longitude,
+              name: params.name,
+            });
+
+          case ActionType.ADD_STOP_GROUP_TO_STOP_CLUSTER:
+            return this.addStopGroupToStopCluster(groundPlaceGpuid, intoGroundPlaceGpuid);
+
+          case ActionType.REMOVE_STOP_GROUP_FROM_STOP_CLUSTER:
+            return this.removeStopGroupFromStopCluster(groundPlaceGpuid, fromGroundPlaceGpuid);
+
+          case ActionType.MOVE_STOP_GROUP:
+            return this.moveStopGroup(groundPlaceGpuid, fromGroundPlaceGpuid, intoGroundPlaceGpuid);
+
+          case ActionType.MOVE_SEGMENT_PROVIDER_STOP:
+            return this.moveSegmentProviderStop(params.segmentProviderStopId, groundPlaceGpuid, intoGroundPlaceGpuid);
+
+          case ActionType.MERGE_STOP_GROUP:
+            return this.mergeStopGroup(groundPlaceGpuid, intoGroundPlaceGpuid);
+
+          case ActionType.MERGE_STOP_CLUSTER:
+            return this.mergeStopCluster(groundPlaceGpuid, intoGroundPlaceGpuid);
+
+          case ActionType.DELETE_STOP_GROUP:
+            return this.deleteStopGroup(groundPlaceGpuid);
+
+          case ActionType.DELETE_STOP_CLUSTER:
+            return this.deleteStopCluster(groundPlaceGpuid);
+
+          default:
+            return;
+        }
+      });
+    } catch (error) {
+      // If there is an error, previous update is reverted
+      this.storageService.setGroundPlaces(cloneGroundPlaces);
+
+      throw new Error(`There is an error inside your GroundPlacesDiffFile. More details: "${error.message}"`);
+    }
+
+    if (isEqual(groundPlaces, cloneGroundPlaces)) {
```

Comment:
When and why could this happen ?

Reply:
When the structure of the **GroundPlacesActionHistory** file is wrong and there is no change inside our groundPlaces object. Like passing an empty array (`[]`) as **GroundPlacesActionHistory** (or anything that won't works and don't update the groundPlaces). 

If there is no errors but no changes, we also need to tell the user that the updates did not work properly I think

Reply:
I think we should never enter this case

Reply:
ok, i've removed this case